docs/madr/decisions/038-meshmetric-profiles.md

```diff
@@ -0,0 +1,371 @@
+```
+
+Or for Consul Grafana dashboards:
+
+```bash
+cat docs/madr/decisions/assets/038/consul-grafana.json | jq 'try .panels[] | try .targets[] | try .expr' | grep -E -o '\benvoy_[a-zA-Z0-9_]+\b' | sort | uniq
+```
+
+All of these metrics combined result in 100 metrics.
+
+By default, Envoy starts with 378 metrics (unfortunately it's not a complete list):
+
+```bash
+docker run --rm -it -p 9901:9901 -p 10000:10000 envoyproxy/envoy:v1.29.1
+curl -s localhost:10000 > /dev/null
+curl -s localhost:9901/stats/prometheus | grep -E -o '\benvoy_[a-zA-Z0-9_]+\b' | sort | uniq | wc -l
+# 378
+```
+
+And datadog lists 990 metrics in total https://github.com/DataDog/integrations-core/blob/master/envoy/metadata.csv and 329 non-legacy ones
+
+```bash
+curl -s https://raw.githubusercontent.com/DataDog/integrations-core/master/envoy/metadata.csv | grep -v Legacy | wc -l
+```
+
+We can build automation on top of this, so when we update dashboards or Envoy changes metrics it emits we know about this and can adjust accordingly
+(just like now official Envoy dashboards - `envoy_cluster_upstream_rq_time_99percentile` have metrics that no longer exist).
+
+As you can see there is no easy way to track everything (Envoy does not by default print all possible metrics)
+so I strongly suggest adding a feature to dynamically (by regex for example) add/remove metrics to/from existing profiles.
+
+#### Behaviour
+
+Profiles can be either static or additive.
+A static profile means you get X metrics with profile Y and you can't combine that with other profiles.
+Additive profiles allow you to mix and match which metrics you need depending on current circumstances (e.g. debugging HDS issue)
+and specifics of the underlying application (e.g. `tcp` app does not need `http` metrics).
+On top of that you can manually include / exclude metrics from there.
+
+##### User defined profiles
+
+We could allow people to define their custom profiles and reference them in `MeshMetric`.
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshMetricsProfile
+metadata:
+  name: my-profile
+  namespace: kuma-system
+spec:
+  default:
+    appendMetricExact:
+      - envoy_cluster_upstream_rq_xx
+      - envoy_http_downstream_rq_xx
+    appendMetricRegex:
+      - .*error.*
+```
+
+##### Static
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  profile:
+    ref: my-profile
+```
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  profile:
+    ref: default # our predefined profile
+```
+
+##### Additive
+
+```yaml
+sidecar:
+  usedOnly: true # true or false
+  appendProfiles:
+    - ref: my-profile
+    - ref: default # our predefined profile
+```
+
+The question here is: do we want our builtin profiles to be defined like that (this possibly means a user can edit them a bit like default policies)
+or should they be hidden from the user (and only referencable)?
+Since we're starting without this it can be skipped for now but in the future this needs reconsideration.
```

Comment:
I think it is not actually written out in the MADR that we push this feature to the future